IPython/utils/_process_common.py

```diff
@@ -70,6 +71,7 @@ def process_handler(cmd, callback, stderr=subprocess.PIPE):
     # On win32, close_fds can't be true when using pipes for stdin/out/err
     close_fds = sys.platform != 'win32'
     p = subprocess.Popen(cmd, shell=isinstance(cmd, py3compat.string_types),
+                         executable=os.environ.get('SHELL'),
```

Comment:
This should only be set if we're using the shell - which, from the line above, is the case if `cmd` is a string (...I don't like this code).

I think we should also only be using this on Unix systems.